File: DataModels/Container.cs
namespace DockerW.DataModels
{
    public class Container
    {
        public string? Id { get; set; }
        public IList<string> Names { get; set; } = new List<string>();
        public string? Image { get; set; }
        public string? ImageId { get; set; }
        public string? Command { get; set; }
        public DateTime Created { get; set; }

        public IList<Port> Ports { get; set; } = new List<Port>();
        //  sizeRw: bigint;
        //  sizeRootFs: bigint;
        public IDictionary<string, string> Labels { get; set; } = new Dictionary<string, string>();
        public string? State { get; set; }
        public string? Status { get;set; }

        //  networkSettings: ISummaryNetworkSettings;
        //  mounts: IMountPoint[];
    }

    public class Port
    {
        public string Ip { get; set; } = String.Empty;   
        public ushort PrivatePort { get; set; }
        public ushort PublicPort { get; set; }
        public string Type { get; set; } = String.Empty;
    }
}

File: ClientApp/src/app/containers/container.ts
export interface Container {
  id: string;
  names: string[];
  image: string;
  imageId: string;
  command: string;
  created: string;
  ports: IPort[];
  sizeRw: bigint;
  sizeRootFs: bigint;
  labels: Map<string, string>;
  state: string;
  status: string;
  networkSettings: ISummaryNetworkSettings;
  mounts: IMountPoint[];
}

export interface IMountPoint {
  type: string;
  name: string;
  source: string;
  destination: string;
  driver: string;
  mode: string;
  rw: boolean;
  propagation: string;
}

export interface IPort {
  ip: string;
  privatePort: number;
  publicPort: number;
  type: string;
}

export interface ISummaryNetworkSettings {
  networks: Map<string, IEndpointSettings>;
}

export interface IEndpointSettings {
  IPAMConfig: EndpointIPAMConfig;
  links: string[];
  aliases: string[];
  networkID: string;
  endpointID: string;
  gateway: string;
  IPAddress: string;
  IPPrefixLen: bigint;
  IPv6Gateway: string;
  globalIPv6Address: string;
  globalIPv6PrefixLen: bigint;
  macAddress: string;
  driverOpts: Map<string, string>;
}

export interface EndpointIPAMConfig {
  IPv4Address: string;
  IPv6Address: string;
  linkLocalIPs: string[];
}